addons/mrp/models/mrp_production.py

```diff
@@ -762,14 +762,17 @@ def _render_note_exception_quantity_mo(rendering_context):
                 order_exceptions.update(order_exception)
                 visited_objects += visited
             visited_objects = self.env[visited_objects[0]._name].concat(*visited_objects)
-            visited_objects |= visited_objects.mapped('move_orig_ids')
-            impacted_pickings = []
+            if visited_objects:
+                visited_objects |= visited_objects.mapped('move_orig_ids')
+            impacted_object = []
             if visited_objects._name == 'stock.move':
-                impacted_pickings = visited_objects.filtered(lambda m: m.state not in ('done', 'cancel')).mapped('picking_id')
+                impacted_object = visited_objects.filtered(lambda m: m.state not in ('done', 'cancel')).mapped('picking_id')
+            elif visited_objects._name == 'purchase.order.line':
```

Comment:
I don't think you sometimes have a PO line (with a real record) as a visited object.